--- 
title: "COVID-19 Live Report"
author: "Enayetur Raheem, Ph.D."
date: "`r Sys.Date()`"
site: bookdown::bookdown_site
output: bookdown::gitbook
documentclass: book
bibliography: [book.bib, packages.bib]
biblio-style: apalike
link-citations: yes
github-repo: raheems/covid19
cover-image: "https://covid19.eraheem.com/images/cover.png"
description: "Live report on COVID-19  pandemic a.k.a. SARS-CoV2 with emphasis on data visualizations and interpretations. The report is focused on current situation in Bangladesh and the aftermath of COVID-19."
---

# About this report {-}

This is a live report about the COVID-19 pandemic also known as SARS-CoV2. 

```{r fig.align='center', echo=FALSE, include=identical(knitr:::pandoc_to(), 'html'),}
knitr::include_graphics('images/cover.png', dpi = NA)
```

## Disclaimer {-}

The information provided here are for educational purposes only. All data have been collected from freely available sources. For details, please check out the Reference section of this report.

This live report is updated a few times daily. So please check back to get the most up to date report. 

## Purpose {-}

The goal of this report is to provide up to date summary of the global situation and compare that with __Bangladesh__.


## Volunteering opportunity {-}

If you have experience with R programming, and have interest in contributing to this project, please reach out to me via email: enayetur.raheem@brfbd.org. Please mention specifically how do you want to contribute. Because I receive too many inquiries, I cannot reply to each an every email. I will only respond if your email has sufficient information to spark my interest. Thanks in advance :)

### What skills are useful {-}

- writing R functions to perform a specific task given a data set
- fitting statistical models
- time series forecasting

## Found an error? {-}

- If you have found an error, please create an issue at https://github.com/raheems/covid19/issues 


<!--chapter:end:index.Rmd-->


# World {#world}


## Daily Incidence, Recovery, Deaths {#incidence-recovery-death}

```{r, echo=FALSE, message=FALSE, cache=FALSE, warning=FALSE}
# source("global_incidence_recovery_deaths.R")
source("covid19_global.R")
source("load_functions.R")
```

__Report last updated:__ `r Sys.Date()`

__Data last refreshed:__ : `r data_last_refreshed`

```{r global-incidence-recovery-deaths, fig.width=8, fig.height=6, echo = FALSE, warning=FALSE}

p <- global %>%
  group_by(date) %>%
  summarise(
    daily_cases = sum(daily_cases)
    , daily_recov = sum(daily_recov)
    , daily_deaths = sum(daily_deaths)
    , daily_recov_death_ratio = (daily_recov/daily_deaths)
  ) %>%
  gather(metric, n, daily_recov, daily_cases, daily_deaths, factor_key = TRUE) %>%
  mutate(
    metric = factor(metric, levels = c("daily_deaths", "daily_recov", "daily_cases"))
  ) %>%
  ggplot(aes(x=date, y=n, color=metric))+
  # geom_point(aes(y=rollmean(n, 7, na.pad=TRUE))) +
  geom_point() +
  geom_smooth() +
  theme_minimal(base_size = 14) +
  theme(legend.position="top") +
  ggtitle("GLOBAL daily incidence, recovery, deaths", data_last_refreshed ) +
  xlab("Date") +
  ylab("Daily cases")+
  labs(
    title= "GLOBAL daily incidence, recovery, deaths  (log scale)",
    caption = "SOURCE: Johns Hopkins Coronavirus Resource Center"
    )
p = p + scale_y_continuous(labels=comma_format(), trans='log10') +
  scale_color_manual(name="Metric", 
                   breaks=c("daily_deaths", 
                                  "daily_recov", "daily_cases"),
                   labels=c("Daily deaths", 
                                  "Daily recovery",
                                  "Daily cases"),
                   values=ggplotColours(n = 3)) +
   guides(fill=guides(colour=guide_legend(ncol=3,nrow=1,byrow=TRUE)))

suppressMessages(print(p))

```


## Case Fatality Rate {#world-case-fatality}

Case fatality rate is calculated by dividing the number of deaths from a specified disease over a defined period of time by the number of individuals diagnosed with the disease during that time; the resulting ratio is then multiplied by 100 to yield a percentage [@cfr_britanica].

$$CFR = \frac{\# Deaths}{\#Infected} \times 100 \,\, per \,\,unit \,\, time$$

```{r, global-case-fatality-rate, fig.width=8, fig.height=6, echo = FALSE, warning=FALSE}

df <- case_fatality(global)

p <- df %>%
  ggplot(aes(x=date, y=n, color=metric))+
  geom_point() +
  geom_smooth() +
  theme_minimal(base_size = 14) +
  theme(legend.position="top") +
  ggtitle("GLOBAL case fatality rate", data_last_refreshed ) +
  xlab("Date") +
  ylab("Case fatality rate (%)")+
  labs(
    title= "GLOBAL case fatality rate",
    caption = "SOURCE: Johns Hopkins Coronavirus Resource Center"
  )
p = p + scale_y_continuous(labels=comma, expand=c(0.5, 0.5)) +
  scale_color_discrete(name="Metric",
                       breaks=c("cum_case_fatality_rate",
                                "daily_case_fatality_rate"),
                       labels=c("Cumulative case fatality",
                                "Daily case fatality")) +
  guides(fill=guides(colour=guide_legend(ncol=3,nrow=1,byrow=TRUE)))

suppressMessages(print(p))

```

### Case Fatality Rate (select countries)


```{r, case-fatality-rate-selected, fig.width=8, fig.height=6, echo = FALSE, warning=FALSE}

country_list = c("Bangladesh", "US", "United Kingdom", "Canada", "France", 
                 "Italy", "India", "Pakistan", "")
 
df <- global %>%
  filter(country %in% country_list, cum_cases >0) %>%
  group_by(country, date) %>%
    summarise(
      cum_case_fatality_rate = (sum(cum_deaths)/sum(cum_cases)) * 100
      , daily_case_fatality_rate = (sum(daily_deaths)/sum(cum_cases)) * 100
    ) %>%
    group_by(country) %>%
  mutate(
   day = row_number()
  ) %>%
  gather(metric, n, cum_case_fatality_rate, daily_case_fatality_rate, factor_key = TRUE)


p <- df %>%
  ggplot(aes(x=day, y=n, color=metric))+
  geom_point() +
  geom_smooth() +
  theme_minimal(base_size = 14) +
  theme(legend.position="top") +
  ggtitle("GLOBAL case fatality rate", data_last_refreshed ) +
  xlab("Day since first case") +
  ylab("Case fatality rate (%)")+
  labs(
    title= "CASE fatality rate (selected countries)",
    caption = "SOURCE: Johns Hopkins Coronavirus Resource Center"
  )
p = p + scale_y_continuous(labels=comma, expand=c(0.5, 0.5)) +
  scale_color_discrete(name="Metric",
                       breaks=c("cum_case_fatality_rate",
                                "daily_case_fatality_rate"),
                       labels=c("Cumulative case fatality",
                                "Daily case fatality")) +
  guides(fill=guides(colour=guide_legend(ncol=3,nrow=1,byrow=TRUE))) +
  facet_wrap( ~ country, ncol = 2)

suppressMessages(print(p))


```

## Confirmed Cases 


```{r, global-confirmed, fig.width=8, fig.height=6, echo = FALSE, warning=FALSE}
# Cumulative Cases since first identified case

# Confirmed for Bangladesh
df_bd <- global %>%
  filter(
    country %in% c('Bangladesh')
  ) %>%
  select(country, date, cum_cases) %>%
  arrange(
    country, date
  ) 

# Top 10 countries
n_world <- global %>%
  filter(country != 'Bangladesh(unoff)' ) %>%
  group_by(country) %>%
  summarise(
    max_n_cases = max(cum_cases)
  ) %>%
  arrange(desc(max_n_cases)) %>%
  top_n(6, max_n_cases)

# Top 10 countries
n_bd <- df_bd %>%
  group_by(country) %>%
  summarise(
    max_n_cases = max(cum_cases)
  )


# Combined n countries
n <- rbind(n_world, n_bd)

# Reset the origin to the first reported cases
case_summary <- global %>%
  filter(country %in% n$country, cum_cases >0) %>%
  arrange(country, date, cum_cases) %>%
  group_by(country) %>%
  mutate(
   day = row_number()
  )
data_last_refreshed = max(case_summary$date)

p <- ggplot(case_summary, aes(x = day, y = cum_cases, color=country)) +
  geom_line(size=1.1) +
  geom_point(data = case_summary[case_summary$country == 'Bangladesh' & ymd(case_summary$date)==data_last_refreshed, ],
             aes(x=day, y=cum_cases), color='red', size=5) +
  geom_text_repel(size=5, aes(label=comma_format()(cum_cases)),
                  function(case_summary) case_summary[ymd(case_summary$date) == data_last_refreshed, ])+
  theme_minimal(base_size = 14) +
  theme(legend.position="top") +
  ggtitle("COVID19 cumulative cases since first case \n(top 6 countries)", data_last_refreshed ) +
  xlab("Days since first confirmed case") +
  ylab("Cumulative cases")+
  labs(caption = "SOURCE: Johns Hopkins Coronavirus Resource Center.")

p + scale_color_discrete(name="Country") +
  scale_y_continuous(labels=comma, trans = 'log10')

# +
#   guides(fill=guides(colour=guide_legend(ncol=1,nrow=1,byrow=TRUE))) +
#   scale_y_continuous(labels = function(x) format(x, scientific = FALSE))



```


```{r, echo=FALSE}

# Summary of cases till date by country
world_summary <- global %>%
  # filter(country != 'Bangladesh(unoff)') %>%
  filter(ymd(date) == data_last_refreshed) %>%
  group_by(country) %>%
  summarise(
    Confirmed = max(cum_cases)
  ) %>%
  top_n(10, Confirmed) %>%
  arrange(desc(Confirmed))

confirmed_bd <- bd_unoff %>%
  filter(ymd(date) == data_last_refreshed) %>%
  group_by(country) %>%
  summarise(
    Confirmed = max(cum_cases)
  ) %>%
  arrange(desc(Confirmed))

# Confirmed (world + Bangladesh)

world_plus_bd <- rbind(world_summary, confirmed_bd)

world_plus_bd %>%
  kable(format = "html", caption=paste("Top 10 countries by confirmed cases as of", data_last_refreshed)) %>%
  kable_styling(position = "center")


```


## Deaths since first death


```{r, global-deaths-since-first-death, fig.width=8, fig.height=7, echo = FALSE, warning=FALSE}
# Cumulative Cases since first identified case

b <- bd_unoff
a <- global %>% filter(country %in% "Bangladesh", cum_deaths > 0) %>%
  select(-c(daily_cases, daily_deaths, daily_recov))

bd <- bind_rows(a,b) %>%
  arrange(country, date) %>%
  mutate(
   day = row_number()
  )

# Top 10 countries
n_world <- global %>%
  group_by(country) %>%
  summarise(
    max_n_cases = max(cum_deaths)
  ) %>%
  arrange(desc(max_n_cases)) %>%
  top_n(6, max_n_cases)


df_world <- global %>%
  filter(country %in% n_world$country, cum_deaths >0) %>%
  select(date, country, cum_cases, cum_deaths, cum_recov)

df_world_bd <- bind_rows(bd, df_world)


# Reset the origin to the first reported cases
case_summary <- df_world_bd %>%
  filter(cum_deaths >0) %>%
  arrange(country, date, cum_deaths) %>%
  group_by(country) %>%
  mutate(
   day = row_number()
  )


max_date_bd_unoff <- case_summary %>%
  filter(country == "Bangladesh(unoff)") %>% group_by(country) %>%
  summarise(date=max(date))

max_date_bd_off <- case_summary %>%
  filter(country == "Bangladesh") %>% group_by(country) %>%
  summarise(date=max(date))


p <- ggplot(case_summary, aes(x = day, y = cum_deaths, color=country)) +
  geom_line(size=1.1) +
  geom_point(data = case_summary[case_summary$country == 'Bangladesh(unoff)' & case_summary$date==max_date_bd_unoff$date, ], aes(x=day, y=cum_deaths), color='red', size=5) +
  geom_point(data = case_summary[case_summary$country == 'Bangladesh' & case_summary$date==max_date_bd_off$date, ], aes(x=day, y=cum_deaths), color='red', size=5) +
  geom_text_repel(size=5, aes(label=country),
                  function(case_summary) case_summary[case_summary$date == max_date_bd_off$date, ])+
  
  theme_minimal(base_size = 14) +
  theme(legend.position="top") +
  ggtitle("COVID19 cumulative deaths since first death", data_last_refreshed ) +
  xlab("Days since first death") +
  ylab("Cumulative deaths")+
  labs(caption = "SOURCE: Johns Hopkins Coronavirus Resource Center. Bangladesh(unofficial) data compiled from newspaper reports.")

p + scale_color_discrete(name="Country") +
  scale_y_continuous(labels=comma_format(), trans = 'log10')

  # +
#   guides(fill=guides(colour=guide_legend(ncol=1,nrow=1,byrow=TRUE))) +
#   scale_y_continuous(labels = function(x) format(x, scientific = FALSE))



```

## Deaths since first confirmed case



```{r, global-deaths-since-first-case, fig.width=8, fig.height=6, echo = FALSE, warning=FALSE}
# Cumulative Cases since first identified case

b <- bd_unoff
a <- global %>% filter(country %in% "Bangladesh", cum_cases > 0) %>%
  select(-c(daily_cases, daily_deaths, daily_recov))

bd <- bind_rows(a,b) %>%
  arrange(country, date) %>%
  mutate(
   day = row_number()
  )

# Top 10 countries
n_world <- global %>%
  group_by(country) %>%
  summarise(
    max_n_cases = max(cum_deaths)
  ) %>%
  arrange(desc(max_n_cases)) %>%
  top_n(6, max_n_cases)


df_world <- global %>%
  filter(country %in% n_world$country, cum_cases >0) %>%
  select(date, country, cum_cases, cum_deaths, cum_recov)

df_world_bd <- bind_rows(bd, df_world)


# Reset the origin to the first reported cases
case_summary <- df_world_bd %>%
  filter(cum_cases >0) %>%
  arrange(country, date, cum_deaths) %>%
  group_by(country) %>%
  mutate(
   day = row_number()
  )


max_date_bd_unoff <- case_summary %>%
  filter(country == "Bangladesh(unoff)") %>% group_by(country) %>%
  summarise(date=max(date))

max_date_bd_off <- case_summary %>%
  filter(country == "Bangladesh") %>% group_by(country) %>%
  summarise(date=max(date))


p <- ggplot(case_summary, aes(x = day, y = cum_deaths, color=country)) +
  geom_line(size=1.1) +
  geom_point(data = case_summary[case_summary$country == 'Bangladesh(unoff)' & case_summary$date==max_date_bd_unoff$date, ], aes(x=day, y=cum_deaths), color='red', size=5) +
  geom_point(data = case_summary[case_summary$country == 'Bangladesh' & case_summary$date==max_date_bd_off$date, ], aes(x=day, y=cum_deaths), color='red', size=5) +
  geom_text_repel(size=5, aes(label=country),
                  function(case_summary) case_summary[case_summary$date == max_date_bd_off$date, ])+
  
  theme_minimal(base_size = 14) +
  theme_minimal(base_family="Siyam Rupali") +
  theme(legend.position="none") +
  ggtitle("COVID19 cumulative deaths since first confirmed case", data_last_refreshed ) +
  xlab("Days since first confirmed case") +
  ylab("Cumulative deaths")+
  labs(caption = "SOURCE: Johns Hopkins Coronavirus Resource Center. Bangladesh(unofficial) data compiled from newspaper reports")

p + scale_color_discrete(name="Country") +
  scale_y_continuous(labels=comma_format(), trans = 'log10') +
  scale_x_continuous(limits=c(0,max(case_summary$day)+5), breaks=seq(0,max(case_summary$day)+5,10))

  # +
#   guides(fill=guides(colour=guide_legend(ncol=1,nrow=1,byrow=TRUE))) +
#   scale_y_continuous(labels = function(x) format(x, scientific = FALSE))



```




```{r, echo=FALSE}

# Summary of cases till date by country
world_summary <- global %>%
  filter(ymd(date) == data_last_refreshed) %>%
  group_by(country) %>%
  summarise(
    Confirmed = max(cum_deaths)
  ) %>%
  top_n(10, Confirmed) %>%
  arrange(desc(Confirmed))

confirmed_bd <- bd_unoff %>%
  filter(ymd(date) == max_date_bd_unoff$date) %>%
  group_by(country) %>%
  summarise(
    Confirmed = max(cum_deaths)
  ) %>%
  arrange(desc(Confirmed))

# Confirmed (world + Bangladesh)

world_plus_bd <- rbind(world_summary, confirmed_bd)

world_plus_bd %>%
  kable(format = "html", caption=paste("Top 10 countries by confirmed deaths as of", data_last_refreshed)) %>%
  kable_styling(position = "center")


```


### Cases and deaths compared on a specific day

```{r, echo=FALSE}
bd_day_indicator = df_world_bd %>% filter(country == 'Bangladesh', 
                                                date == max(date)) %>%
  pull(day)
```
Bangladesh has entered into day `r bd_day_indicator` since first confirmed case.


```{r, echo=FALSE}

# Summary of deaths till date by country on the specific day

bd_world_summary <- df_world_bd %>%
  group_by(country) %>%
  mutate(
   day = row_number()
  ) %>%
  filter(day == bd_day_indicator) %>%
  group_by(country) %>%
  summarise(
    `Total cases` = max(cum_cases),
    `Total deaths` = max(cum_deaths),
    Country = country
  ) %>%
  select(Country, `Total cases`, `Total deaths`) %>%
  arrange(desc(`Total deaths`))

bd_world_summary %>%
  kable(format = "html", caption=paste0("Bangladesh in the world: total cases and deaths compared on day ", bd_day_indicator)) %>%
  kable_styling(position = "center") %>%
  row_spec(which(bd_world_summary$Country %in% c('Bangladesh', 'Bangladesh(unoff)')), bold = T, color = "white", background = "#f49a86")

```

## Recovery-to-death ratio {#recovery-death-ratio}


The recovery to deaths ratio (RDR) is calculated as

$$
RDR = \frac{\# Cumulative \,\, daily \,\, recovered}{\# Cumulative \,\, daily \,\, deaths}
$$


__How to interpret__: The ratio of cumulative number of patients recovered divided by the cumulative number of deaths over time can tell the proportion of people recovering compared to dying. An upward trend may indicate more recovery than deaths. 


```{r, recovery-to-death-ratio-by-country, fig.width=8, fig.height=6, echo = FALSE, warning=FALSE}

# Global top 5 countries by deaths --------------

# Top n countries
n_world <- global %>%
  group_by(country) %>%
  summarise(
    deaths = max(cum_deaths)
    , recovered = max(cum_recov)
  ) %>%
  arrange(country) %>%
  top_n(5, deaths)

# Combined n countries
n <- n_world

df <- global %>%
  filter(country != "Bangladesh(unoff)") %>%
  mutate(
    # cum_recov_death_ratio = cum_recov/cum_deaths
    cum_recov_death_ratio = cum_recov/cum_deaths
  ) %>%
  filter(is.finite(cum_recov_death_ratio)) %>%
  filter(country %in% n$country, cum_recov_death_ratio >= 0) %>%
  arrange(country, date, cum_recov_death_ratio) %>%
  group_by(country) %>%
  mutate(
    day = row_number()
  )
   p <- ggplot(df, aes(x = day, y = cum_recov_death_ratio, color=country)) +
      geom_point(size=1) + geom_smooth() +
     geom_text_repel(size=5, aes(label=country),
                     function(df) df[ymd(df$date) == data_last_refreshed, ])+
     theme_minimal(base_size = 14)

  p <-  p + ggtitle("Recovery-to-death ratio \n(Top 5 countries by death, excluding China)", data_last_refreshed ) +
      xlab("Days since first death") +
      ylab("Cumulative recovery / cumulative deaths") +
      theme(legend.position="top") +
      labs(caption = "SOURCE: Johns Hopkins Coronavirus Resource Center") +
      scale_color_discrete(name="Country")
  
suppressMessages(print(p))


```


### Recovery-to-death ratio (South and South-East Asia)


```{r, recovery-to-death-ratio-south-asia, fig.width=8, fig.height=6, echo = FALSE, warning=FALSE}

country_list = c("Bangladesh", "India", "Singapore", "Nepal", "Pakistan",
                   "Bhutan", "Sri Lanka", "Malaysia", "Maldives", 
                   "Indonesia", "Thailand")

# Top n countries
n_world <- global %>%
  filter(country %in% country_list) %>%
  group_by(country) %>%
  summarise(
    deaths = max(cum_deaths)
    , recovered = max(cum_recov)
  ) %>%
  arrange(country) %>%
  top_n(10, deaths)

# Combined n countries
n <- n_world

df <- global %>%
  mutate(
    cum_recov_death_ratio = cum_recov/cum_deaths
  ) %>%
  filter(country %in% country_list) %>%
  filter(is.finite(cum_recov_death_ratio)) %>%
  filter(country %in% n$country, cum_recov_death_ratio >= 0) %>%
  arrange(country, date, cum_recov_death_ratio) %>%
  group_by(country) %>%
  mutate(
    day = row_number()
  )

data_last_refreshed = max(df$date)
# plot
p <- ggplot(df, aes(x = day, y = cum_recov_death_ratio, color=country)) +
  geom_point(size=1) + geom_smooth() +
  geom_text_repel(size=5, aes(label=country),
                  function(df) df[ymd(df$date) == data_last_refreshed, ])+
  theme_minimal(base_size = 14)

p <-  p + ggtitle("Recovery-to-death ratio \n(South Asian countries)", data_last_refreshed ) +
  xlab("Days since first death") +
  ylab("Cumulative recovery / cumulative deaths") +
  theme(legend.position="top") +
  labs(caption = "SOURCE: Johns Hopkins Coronavirus Resource Center") +
  scale_color_discrete(name="Country")

suppressMessages(print(p))

```

### Recovery-to-death ratio (Middle East)


```{r, recovery-to-death-ratio-middle-east, fig.width=8, fig.height=6, echo = FALSE, warning=FALSE}

country_list = c("Saudi Arabia", "Iraq", "Iran", "Israel",  "United Arab Emirates")

# Top n countries
n_world <- global %>%
  filter(country %in% country_list) %>%
  group_by(country) %>%
  summarise(
    deaths = max(cum_deaths)
    , recovered = max(cum_recov)
  ) %>%
  arrange(country) %>%
  top_n(10, deaths)

# Combined n countries
n <- n_world

df <- global %>%
  mutate(
    cum_recov_death_ratio = cum_recov/cum_deaths
  ) %>%
  filter(country %in% country_list) %>%
  filter(is.finite(cum_recov_death_ratio)) %>%
  filter(country %in% n$country, cum_recov_death_ratio >= 0) %>%
  arrange(country, date, cum_recov_death_ratio) %>%
  group_by(country) %>%
  mutate(
    day = row_number()
  )

# plot
p <- ggplot(df, aes(x = day, y = cum_recov_death_ratio, color=country)) +
  geom_point(size=1) + geom_smooth() +
  geom_text_repel(size=5, aes(label=country),
                  function(df) df[ymd(df$date) == data_last_refreshed, ])+
  theme_minimal(base_size = 14)

p <-  p + ggtitle("Recovery-to-death ratio \n(Middle Eastern countries)", data_last_refreshed ) +
  xlab("Days since first death") +
  ylab("Cumulative recovery / cumulative deaths") +
  theme(legend.position="top") +
  labs(caption = "SOURCE: Johns Hopkins Coronavirus Resource Center") +
  scale_color_discrete(name="Country")

suppressMessages(print(p))

```


## 3-day % changes {#perc-change-top6}



```{r, 3day-perc-change, fig.width=8, fig.height=9, echo = FALSE, warning=FALSE}

# Global top 6 countries by deaths --------------

# Top 6 countries by deaths
global_top_deaths <- global %>%
  filter(!country %in% c('China')) %>%
  group_by(country) %>%
  summarise(
    max_n_cases = max(cum_deaths)
  ) %>%
  arrange(desc(max_n_cases)) %>%
  top_n(6, max_n_cases)

p <- global_gr_lagx(global, global_top_deaths$country, 3)
p <- p + theme(legend.position="none") +
  ggtitle("COVID19 3-day % change in select metrics \n(Top 6 countries by death, excluding China)", data_last_refreshed ) +
  xlab("Days since first death") +
  ylab("% change") +
  theme(legend.position="bottom") +
  labs(caption = "SOURCE: Johns Hopkins Coronavirus Resource Center")

p = p + scale_color_discrete(name="Metric",
                         breaks=c("case_gr_rate", "death_gr_rate"),
                         labels=c("New cases", "Deaths")) +
  guides(fill=guides(colour=guide_legend(ncol=2,nrow=1,byrow=TRUE))) +
  facet_wrap( ~ country, ncol = 2) +
  ylim(c(-20, 300))
p

```


The percent change is calculated as 

$$
PC = \frac{New - Original}{Original} * 100
$$
where `New` is the number of cases on a given date, and `Original` is the number 3 days prior to that. 

__How to interpret__: The 3-day percent change tells how much does a metric have changed (%) compared to what it was 3-days ago. A downward trending line suggests a slow growth.


<!--chapter:end:01-world.Rmd-->


# USA {#usa}

__Report last updated:__ `r Sys.Date()`

__Data last refreshed:__ : `r data_last_refreshed`


```{r, echo=FALSE, message=FALSE, cache=FALSE}
source("covid19_usa.R")
```


## Overall trend {#us-overall}

```{r, fig.width=8, fig.height=6, echo = FALSE, warning=FALSE}

p <- usa %>%
  group_by(date) %>%
  summarise(
    cum_cases = sum(cum_cases)
    , cum_deaths = sum(cum_deaths)
  ) %>% 
  gather(metric, n, cum_cases, cum_deaths, factor_key = TRUE) %>%
  mutate(
    metric = factor(metric, levels= c("cum_deaths", "cum_cases"))
  ) %>%
  ggplot(aes(x = date, y = n, color=metric)) +
  geom_line(size=1.1) +
  theme_minimal(base_size = 14) +
  theme(legend.position="top") +
  ggtitle("COVID19 USA Overall", data_last_refreshed ) +
  xlab("Days since first confirmed case") +
  ylab("Cumulative cases") +
  labs(caption = "SOURCE: New York Times Coronavirus Dataset.")

 p = p + scale_y_continuous(labels=comma_format(), trans = 'log10')

 p = p + scale_color_discrete(name="Metric",
                         breaks=c("cum_deaths", "cum_cases"),
                         labels=c("Deaths", "New Cases"))
 
 suppressWarnings(print(p))
 

```


## Confirmed Cases 


```{r, fig.width=8, fig.height=6, echo = FALSE, warning=FALSE}
# Cumulative Cases since first identified case

# Top n
us_topn_death <- usa %>%
  group_by(state) %>%
  summarise(
    max_n_cases = max(cum_deaths)
  ) %>%
  arrange(desc(max_n_cases)) %>%
  top_n(10, max_n_cases)


p <- usa %>%
  filter(cum_cases >0, state %in% us_topn_death$state) %>%
  group_by(state) %>%
  mutate(
   day = row_number()
  ) %>%
  ggplot(aes(x = day, y = cum_cases, color=state)) +
  geom_line(size=1.1) +
   geom_text_repel(size=5, aes(label=state),
                  function(us_topn_death) us_topn_death[ymd(us_topn_death$date) == data_last_refreshed, ])+
  theme_minimal(base_size = 14) +
  theme(legend.position="top") +
  ggtitle("COVID19 USA New Cases by State", data_last_refreshed ) +
  xlab("Days since first confirmed case") +
  ylab("Cumulative cases") +
  labs(caption = "SOURCE: New York Times Coronavirus Dataset.")

 p = p + scale_y_continuous(labels=comma_format(), trans = 'log10')

 p = p + scale_color_discrete(name="Metric",
                         breaks=c("cum_deaths", "cum_cases"),
                         labels=c("Deaths", "New Cases"))
 
 suppressWarnings(print(p))


```


```{r, cases-by-usa-states, fig.width=8, fig.height=6, echo = FALSE, warning=FALSE}

# Top n
us_topn_death <- usa %>%
  group_by(state) %>%
  summarise(
    max_n_cases = max(cum_deaths)
  ) %>%
  arrange(desc(max_n_cases)) %>%
  top_n(20, max_n_cases)


usa %>%
  filter(cum_cases >0, state %in% us_topn_death$state) %>%
  group_by(state) %>%
  summarise(
   n = max(cum_cases)
  ) %>%
  ggplot(aes(x = reorder(state, n), y = n), fill=state) +
  geom_bar(stat = 'identity') +
  theme(axis.text.x=element_text(angle=90,hjust=1)) +
  scale_y_continuous(labels = comma_format())+
  ggtitle("COVID19 USA cases by state", data_last_refreshed ) +
  xlab("States") +
  ylab("Cumulative cases") +
  labs(caption = "SOURCE: New York Times Coronavirus Dataset.")

```

```{r, echo=FALSE}

# USA cases by state
usa %>%
  group_by(state) %>%
  summarise(
    max_n_cases = max(cum_cases)
  ) %>%
  arrange(desc(max_n_cases)) %>%
  top_n(15, max_n_cases)%>%
  transmute(
    State = state
    , `Total cases` = max_n_cases
  ) %>%
  kable(caption = paste("Totaly cases for top 10 US States as of ", data_last_refreshed))
```



## Deaths

```{r, fig.width=8, fig.height=6, echo = FALSE, warning=FALSE}
# Cumulative Cases since first identified case

# Top n
us_topn_death <- usa %>%
  group_by(state) %>%
  summarise(
    max_n_cases = max(cum_deaths)
  ) %>%
  arrange(desc(max_n_cases)) %>%
  top_n(10, max_n_cases)


dfw <- usa %>%
  filter(cum_deaths >0, state %in% us_topn_death$state) %>%
  group_by(state) %>%
  mutate(
   day = row_number()
  )

p <- ggplot(dfw, aes(x = day, y = cum_deaths, color=state)) +
  geom_line(size=1.1) +
  geom_text_repel(size=5, aes(label=cum_deaths),
                  function(dfw) dfw[ymd(dfw$date) == data_last_refreshed, ])+
  theme_minimal(base_size = 14) +
  theme(legend.position="top") +
  ggtitle("COVID19 USA Deaths by State", data_last_refreshed ) +
  xlab("Days since first death") +
  ylab("Cumulative cases") +
  labs(caption = "SOURCE: New York Times Coronavirus Dataset.")

 p <- p + scale_y_continuous(labels=comma_format(), trans ='log10')

 # p = p + scale_color_discrete(name="Metric",
 #                         breaks=c("cum_deaths", "cum_cases"),
 #                         labels=c("Deaths", "New Cases"))
 
 suppressWarnings(print(p))

```


```{r, deaths-by-usa-states, fig.width=8, fig.height=6, echo = FALSE, warning=FALSE}
# Deaths histogram by state

# Top n
us_topn_death <- usa %>%
  group_by(state) %>%
  summarise(
    max_n_cases = max(cum_deaths)
  ) %>%
  arrange(desc(max_n_cases)) %>%
  top_n(20, max_n_cases)


usa %>%
  filter(cum_cases >0, state %in% us_topn_death$state) %>%
  group_by(state) %>%
  summarise(
   n = max(cum_deaths)
  ) %>%
  ggplot(aes(x = reorder(state, n), y = n), fill=state) +
  geom_bar(stat = 'identity') +
  theme(axis.text.x=element_text(angle=90,hjust=1)) +
  scale_y_continuous(labels = comma_format()) +
  ggtitle("COVID19 USA Deaths by State", data_last_refreshed ) +
  xlab("States") +
  ylab("Cumulative cases") +
  labs(caption = "SOURCE: New York Times Coronavirus Dataset.")

```

```{r, echo=FALSE}
# Death table USA by state
us_topn_death %>%
  transmute(
    State = state
    , `Total cases` = max_n_cases
  ) %>%
  kable(caption = paste("Totaly deaths by State as of ", data_last_refreshed))
```

## Daily Deaths {#usa-daily-deaths}


```{r, deaths-usa-daily, fig.width=8, fig.height=6, echo = FALSE, warning=FALSE}

# Deaths daily histogram

# Top n
us_daily_deaths <- usa %>%
  group_by(date) %>%
  summarize(
    total_deaths = max(cum_deaths)
  ) %>%
  mutate(
    daily_deaths = c(0, diff(total_deaths))
  )

a0 <- us_daily_deaths %>%
  ggplot(aes(x=date, y = daily_deaths)) +
  geom_bar(stat = 'identity') + geom_smooth(se=F) +
  theme(axis.text.x=element_text(angle=90,hjust=1)) +
  scale_y_continuous(labels = comma_format()) +
  ggtitle("COVID19 USA daily deaths", data_last_refreshed ) +
  xlab("Date") +
  ylab("Daily deaths") +
  labs(caption = "SOURCE: New York Times Coronavirus Dataset.")

suppressMessages(print(a0))

```


```{r, deaths-usa-ny-daily, fig.width=8, fig.height=6, echo = FALSE, warning=FALSE}

# Deaths daily histogram for New York

# Top n
ny_daily_deaths <- usa %>%
  filter(state == 'New York') %>%
  group_by(date) %>%
  summarize(
    total_deaths = max(cum_deaths)
  ) %>%
  mutate(
    daily_deaths = c(0, diff(total_deaths))
  )

a2 <- ny_daily_deaths %>%
  ggplot(aes(x=date, y = daily_deaths)) +
  geom_bar(stat = 'identity') +
  geom_smooth(se=F) +
  theme(axis.text.x=element_text(angle=90,hjust=1)) +
  scale_y_continuous(labels = comma_format()) +
  ggtitle("COVID19 USA daily deaths for New York", data_last_refreshed ) +
  xlab("Date") +
  ylab("Daily deaths") +
  labs(caption = "SOURCE: New York Times Coronavirus Dataset.")

suppressMessages(print(a2))

```


## Daily Deaths excluding NY {#usa-daily-deaths-excl-ny}


```{r, deaths-usa-daily-excl-ny, fig.width=8, fig.height=6, echo = FALSE, warning=FALSE}

# Deaths daily histogram

# Top n
us_daily_deaths_excl_ny <- usa %>%
  filter(state != 'New York') %>%
  group_by(date) %>%
  summarize(
    total_deaths = max(cum_deaths)
  ) %>%
  mutate(
    daily_deaths = c(0, diff(total_deaths))
  )

a1 <- us_daily_deaths_excl_ny %>%
  ggplot(aes(x=date, y = daily_deaths)) +
  geom_bar(stat = 'identity') +
  # geom_smooth(aes(x=date,y=daily_deaths), se=F, method="loess", show_guide = FALSE,lwd=0.7) +
  geom_smooth(se=F) +
  theme(axis.text.x=element_text(angle=90,hjust=1)) +
  scale_y_continuous(labels = comma_format()) +
  ggtitle("COVID19 USA daily deaths excluding NY", data_last_refreshed ) +
  xlab("Date") +
  ylab("Daily deaths") +
  labs(caption = "SOURCE: New York Times Coronavirus Dataset.")

suppressMessages(print(a1))
```


<!--chapter:end:02-usa.Rmd-->

# Bangladesh {#bangladesh}

```{r, echo=FALSE, message=FALSE, warning=FALSE}
# source("covid19_bd.R")
source("covid19_global.R")
library(incidence)
```

This report is created based on a publicly available data. The data are available on a Google Sheet. Please see the Data Source section for link. 
Please note that publicly available data are __not official__ and __MAY BE UNRELIABLE__. I personally did not verify them. I am using them for educational purposes. Use at your discretion.

## Epidemic curve and search for peak

The epidemic curve for Bangladesh and a tentative peak. The red line in the plot indicates a potential peak based on available data. IF the peak (red line) is at the far right, then we may not have reached a peak yet. We must have sustained decrease of incidences to be sure about any peak. 


```{r, covid19-bangladesh-epidemic-curve-peak, fig.width=8, fig.height=6, echo = FALSE, warning=FALSE}
# BD unofficial cases
date_cut_off = ymd("2020-05-10")
date_data_refreshed = max(bd_unoff$date)

bd_unoff_cases <- bd_unoff %>%
  mutate(
    country = "Bangladesh(unoff)"
    , date = as.Date(date, "%d/%m/%Y")
    , daily_cases = c(0, diff(cum_cases))
  ) %>%
  # filter(date <= date_cut_off) %>%
  select(date, country, cum_cases, daily_cases)
bd_incidence_object <- bd_unoff_cases %>% uncount(daily_cases)
# Crate incidence object using incidence package
bd_incidence_object <- incidence::incidence(bd_incidence_object$date)
# Finding peak
bd_incidence_peak <- incidence::find_peak(bd_incidence_object)
plot(bd_incidence_object) + 
  geom_vline(xintercept = bd_incidence_peak, col = "red", 
             lty = 2, size=1) +  
  labs(title = "Daily unofficial new infections in Bangladesh ", 
       subtitle = "(Red dashed line indicates date of peak infections)") + theme(text = element_text(size = 12))
```


We fit a log linear model to estimate the doubling time (growh phase) and halving time (decline phase, when that happens). The estimated fitted line (solid line) with 95\% confidence interval (dotten lines) are superimposed on the epidemic curve below.


```{r, covid19-bangladesh-epidemic-curve-fitted, fig.width=8, fig.height=6, echo = FALSE, warning=FALSE}
# Fitting the log-linear models to the growth section and the decay section
# bd_incidence_fit <- incidence::fit(bd_incidence_object, split = bd_incidence_peak)
bd_incidence_fit <- incidence::fit(bd_incidence_object)
# plot the incidence data and the model fit
plot(bd_incidence_object) %>% add_incidence_fit(bd_incidence_fit) + 
  labs(title = "Observed and fitted incidence of COVID-19", 
       subtitle = "Bangladesh, 2020") + 
  theme(text = element_text(size = 12))
```

The current doubling time as of `r today()` is `r round(get_info(bd_incidence_fit, "doubling"), 1)` days with a 95% confidence interval of (`r round(get_info(bd_incidence_fit, "doubling.conf"), 1)`) days. 

### Estimating the Overall Reproduction Number 

The log-linear model estimates the $R_0$ as follows:

```{r, echo=FALSE}

# Getting the R0 from log-linear model ----

library(distcrete)
library(epitrix)

mu <- 7.5  # days
sigma <- 3.4  # days
param <- gamma_mucv2shapescale(mu, sigma/mu)

w <- distcrete("gamma", interval = 1, shape = param$shape, scale = param$scale, w = 0)

# growth_R0 <- lm2R0_sample(bd_incidence_fit$before$model, w)
growth_R0 <- lm2R0_sample(bd_incidence_fit$model, w)
```

```{r, echo=FALSE}

hist(growth_R0, col = "grey", border = "white", main = "Distribution of R0")
summary(growth_R0)

```


### Estimating the Instantaneous Reproduction Number ($R_e$)

The estimation of effective reproduction number involves the distribution of serial interval (SI). Based on the literature (REF to be added), a gamma distribution with mean 7.5 and standard deviation 3.5

```{r, echo=FALSE, message=FALSE, warning=FALSE}

library(EpiEstim)

# custom results plotting function to avoid the ugly
# TableGrob messages returned by the plotting function in the
# EpiEstim package
plot_Ri <- function(estimate_R_obj) {
  p_I <- plot(estimate_R_obj, "incid", add_imported_cases = TRUE)  # plots the incidence
  p_SI <- plot(estimate_R_obj, "SI")  # plots the serial interval distribution
  p_Ri <- plot(estimate_R_obj, "R")
  return(gridExtra::grid.arrange(p_I, p_SI, p_Ri, ncol = 2))
}

bd_confirmed_cases <- bd_unoff_cases %>%
  mutate(
    daily_cases = c(0, diff(cum_cases))
  )
  
  
bd_res_parametric_si <- 
  estimate_R(bd_confirmed_cases$daily_cases, 
             method = "parametric_si", 
             config = make_config(list(mean_si = 7.5, std_si = 3.5))
)

plot_Ri(bd_res_parametric_si)
```


### Simulating the $R_e$

Now we perform a simulation of different SI distribution to obtain the $R_e$. For this, we vary the parameters of gamma distribution and obtain the following.


```{r, echo=TRUE, message=FALSE, warning=FALSE}
# Recalculating with varying SI distribution parameters 
# we get these results:

config = make_config(list(mean_si = 7.5, std_mean_si = 3, 
                                       min_mean_si = 2, max_mean_si =10.0,
                                       std_si = 3.5, std_std_si = 1,
                                       min_std_si = 0.5, max_std_si = 4))
bd_res_uncertain_si <- 
  estimate_R(bd_confirmed_cases$daily_cases, method = "uncertain_si", 
             config = config)

plot_Ri(bd_res_uncertain_si)

```

```{r, echo=FALSE}
# print the last 7 days of data
bd_res_uncertain_si$R %>%
  tail(7) %>%
  select(t_start, t_end, `Mean(R)`, `Std(R)`,  `Quantile.0.025(R)`, `Quantile.0.975(R)`) %>%
  kable(format = "html", caption=paste("Instantaneous Effective Reproduction Number by Week as of ", date_data_refreshed)) %>%
  kable_styling(position = "center")

R_CI = tail(bd_res_uncertain_si$R, 1)
R_mean = round(as.numeric(R_CI["Mean(R)"]), 2)
R_LCI = round(as.numeric(R_CI["Quantile.0.025(R)"]), 2)
R_UCI = round(as.numeric(R_CI["Quantile.0.975(R)"]), 2)
```

The effective reproduction number based on the last week's incidences show an average of `r R_mean` with a 95\% confidence interval (`r R_LCI`, `r R_UCI`).

### Cumulative Deaths (official and unofficial)

```{r, covid19-bangladesh-deaths, fig.width=8, fig.height=7, echo = FALSE, warning=FALSE}
b <- bd_unoff
a <- global %>% filter(country %in% "Bangladesh") %>%
  select(-c(daily_cases, daily_deaths, daily_recov), cum_cases)
bd <- bind_rows(a,b) %>%
  filter(cum_cases >0 ) %>%
  arrange(country, date) %>%
  mutate(
   day = row_number()
  )
  
# Plot
p <- ggplot(bd , aes(x = day, y = cum_deaths, color=country)) +
  geom_line(size=1.1) +
  # geom_text_repel(size=5, aes(label=country),
  #                 function(a) a[ymd(a$date) == max(a$date), ])+
  theme_minimal(base_size = 14) +
  theme(legend.position="top") +
  ggtitle("Bangladesh: COVID19 cumulative deaths",
          data_last_refreshed ) +
  xlab("Day since first confirmed case") +
  ylab("Cumulative deaths") +
  theme(legend.position="top") +
  labs(caption = "SOURCE: Bangladesh(unofficial) data compiled from newspaper reports")
p + scale_y_continuous(labels=comma, trans = "log10") 
  # guides(fill=guides(colour=guide_legend(ncol=2,nrow=1,byrow=TRUE)))
```


```{r, echo=FALSE}
bdsummary <- bd %>%
  filter(date == max(date)) %>% 
  group_by(country) %>%
  summarise(
    `Total Deaths` = max(cum_deaths)
  ) %>%
  arrange(desc(`Total Deaths`))
  
bdsummary %>%
  kable(format = "html", caption=paste("Total number of deaths as of ", max(bd$date))) %>%
  kable_styling(position = "center")
```

## Bangladesh in South Asia

### Infection

```{r, incidences-bd-south-asia, fig.width=8, fig.height=7, echo = FALSE, warning=FALSE, message=FALSE}
# source("covid19_deaths.R")
# 
# bd_cases <- bd_unoff %>% filter(country == "Bangladesh(unoff)")
#   select(country, date, cum_cases)
df_south_asia <- global %>%
  filter(
    country %in% c("Bangladesh", "India", "Singapore", "Nepal", "Pakistan",
                   "Bhutan", "Sri Lanka", "Malaysia",
                   "Indonesia")
  ) %>%
  select(country, date, cum_cases) %>%
  group_by(country, date) %>%
  summarize(
    cum_cases = sum(cum_cases)
  )
# Join with BD data
# df_south_asia <- bind_rows(df_south_asia, bd_cases)
n_south_asia <- df_south_asia %>%
  group_by(country) %>%
  summarise(
    max_n = max(cum_cases)
  ) %>%
  arrange(desc(max_n)) %>%
  top_n(10, max_n)
# Combined n countries
# n_south_ais <- rbind(n_world, n_bd)
cases_south_asia <- df_south_asia %>%
  filter(country %in% n_south_asia$country, cum_cases >0) %>%
  arrange(country, cum_cases) %>%
  group_by(country) %>%
  mutate(
    day = row_number()
  )
```

```{r, covid19-cases-south-asia, fig.width=8, fig.height=7, echo = FALSE, warning=FALSE, message=FALSE}
date_data_refreshed = max(cases_south_asia$date)
# Plot
ggplot(cases_south_asia, aes(x = day, y = cum_cases, color=country)) +
  geom_line(size=1.1) +
  geom_point(data = cases_south_asia[cases_south_asia$country == 'Bangladesh' & cases_south_asia$date==max(cases_south_asia$date), ], aes(x=day, y=cum_cases), color='red', size=5) +
  geom_text_repel(size=5, aes(label=country),
                  function(cases_south_asia) cases_south_asia[cases_south_asia$date == max(cases_south_asia$date), ])+
  theme_minimal(base_size = 14) +
  theme(legend.position = "none") +
  # theme_void() + 
  theme(legend.position="none") +
  ggtitle( paste("COVID19 cumulative incidences since first incidence \n(Bangladesh's peers)\n", date_data_refreshed)) +
  xlab("Days since first incidence") +
  ylab("Cumulative cases") +
  labs(caption = "SOURCE: Johns Hopkins Coronavirus Resource Center.")+
  scale_y_continuous(labels=comma, trans = "log10")
```


```{r, echo=FALSE}
# Summary of Infections till date by country
d <- bd %>%
  select(country, date, cum_cases)
df_south_asia <- global %>%
  filter(
    country %in% c("India", "Singapore", "Nepal", "Pakistan",
                   "Bhutan", "Sri Lanka", "Malaysia", 
                   "Indonesia")
  ) %>%
  select(country, date, cum_cases) %>%
  group_by(country, date) %>%
  summarize(
    cum_cases = sum(cum_cases)
  )
df_south_asia <- bind_rows(df_south_asia, d)
n_south_asia <- df_south_asia %>%
  group_by(country) %>%
  summarise(
    max_n = max(cum_cases)
  ) %>%
  arrange(desc(max_n)) %>%
  top_n(20, max_n)
sa_summary_cases <- df_south_asia %>%
  group_by(country) %>%
  summarise(
    `Total Cases` = max(cum_cases)
  ) %>%
  arrange(desc(`Total Cases`))
sa_summary_cases %>%
  kable(format = "html", caption=paste("Total number of cases as of ", date_data_refreshed)) %>%
  kable_styling(position = "center")
```

### Deaths

```{r, southasia-deaths, fig.width=8, fig.height=7, echo = FALSE, warning=FALSE, message=FALSE}
d <- bd %>%
  select(country, date, cum_cases, cum_deaths)
df_south_asia <- global %>%
  select(country, date, cum_deaths, cum_cases) %>%
  filter(
    country %in% c("India", "Singapore", "Nepal", "Pakistan",
                   "Bhutan", "Sri Lanka", "Malaysia", 
                   "Indonesia"),
    cum_cases > 0
  ) %>%
  group_by(country, date) %>%
  summarize(
    cum_deaths = sum(cum_deaths),
    cum_cases = sum(cum_cases)
  )
df_south_asia <- bind_rows(df_south_asia, d)
n_south_asia <- df_south_asia %>%
  group_by(country) %>%
  summarise(
    max_n = max(cum_deaths)
  ) %>%
  arrange(desc(max_n)) %>%
  top_n(10, max_n)
# Combined n countries
# n_south_ais <- rbind(n_world, n_bd)
deaths_south_asia <- df_south_asia %>%
  filter(country %in% n_south_asia$country) %>%
  arrange(country, cum_deaths) %>%
  group_by(country) %>%
  mutate(
    day = row_number()
  )
```


```{r, covid19-death-south-asia, fig.width=8, fig.height=7, echo = FALSE, warning=FALSE, message=FALSE}
max_date_bd_unoff <- deaths_south_asia %>%
  filter(country == "Bangladesh(unoff)") %>% group_by(country) %>%
  summarise(date=max(date))
max_date_bd_off <- deaths_south_asia %>%
  filter(country == "Bangladesh") %>% group_by(country) %>%
  summarise(date=max(date))
# Plot
ggplot(deaths_south_asia, aes(x = day, y = cum_deaths, color=country)) +
  geom_line(size=1.1) +
  geom_point(data = deaths_south_asia[deaths_south_asia$country == 'Bangladesh' & deaths_south_asia$date==max_date_bd_off$date, ], aes(x=day, y=cum_deaths), color='red', size=5) +
  geom_point(data = deaths_south_asia[deaths_south_asia$country == 'Bangladesh(unoff)' & deaths_south_asia$date==max_date_bd_unoff$date, ], aes(x=day, y=cum_deaths), color='red', size=5) +
  geom_text_repel(size=5, aes(label=country),
                  function(deaths_south_asia) deaths_south_asia[deaths_south_asia$date == max_date_bd_off$date, ])+
  # theme_minimal(base_size = 14, base_family="Siyam Rupali") +
  theme_minimal(base_size = 14) +
  theme(legend.position = "none") +
  # theme_void() + 
  theme(legend.position="none") +
  ggtitle( paste("COVID19 cumulative deaths since first incidence \n(Bangladesh's peers)", date_data_refreshed)) +
  xlab("Days since first incidence") +
  ylab("Cumulative deaths") +
  labs(caption = "SOURCE: Johns Hopkins Coronavirus Resource Center. Bangladesh(unoff) data compiled from news reports")+
  scale_y_continuous(labels=comma, trans = "log10")
```


```{r, echo=FALSE}
# Summary of deaths till date by country
sa_summary <- deaths_south_asia %>%
  filter(date == max(date)) %>%
  group_by(country) %>%
  summarise(
    `Total Deaths` = max(cum_deaths)
  ) %>%
  arrange(desc(`Total Deaths`))
sa_summary %>%
  kable(format = "html", caption=paste("Total number of deaths as of ", date_data_refreshed)) %>%
  kable_styling(position = "center")
```


### Cases and deaths compared on a specific day

```{r, echo=FALSE}
bd_day_indicator = df_world_bd %>% filter(country == 'Bangladesh', 
                                                date == max(date)) %>%
  pull(day)
```
Bangladesh has entered into day `r bd_day_indicator` since first confirmed case.


```{r, echo=FALSE}
# Summary of deaths till date by country on the specific day
bd_sa_summary <- df_south_asia %>%
  group_by(country) %>%
  mutate(
   day = row_number()
  ) %>%
  filter(day == bd_day_indicator) %>%
  group_by(country) %>%
  summarise(
    `Total cases` = max(cum_cases),
    `Total deaths` = max(cum_deaths),
    Country = country
  ) %>%
  select(Country, `Total cases`, `Total deaths`) %>%
  arrange(desc(`Total deaths`))
bd_sa_summary %>%
  kable(format = "html", caption=paste0("Bangladesh and its peers: total cases and deaths compared on day ", bd_day_indicator)) %>%
  kable_styling(position = "center") %>%
  row_spec(which(bd_sa_summary$Country %in% c('Bangladesh', 'Bangladesh(unoff)')), bold = T, color = "white", background = "#f49a86")
```

<!--chapter:end:03-bangladesh.Rmd-->


# Projections {#projection}

```{r, echo=FALSE, message=FALSE, warning=FALSE}
# source("covid19_bd.R")
# source("covid19_global.R")
```

__Report last updated:__ `r Sys.Date()`


- In this udpate, second round of prjections (Run 2) have been added. Please scroll down to find the projections. As usual, the last two current values were not used for fitting the model. Also, the model is conservative/pessimistic i.e., it slightly overestimates new cases. 

- In addition to that, a new paramter to control the `run_date` was added to the `fit_seir()` function. See the code for more details.

## Projection for Bangladesh (unofficial) Run1  {#projection-bangladesh-run1}

__Run date__: 17 April 2020

This is a dynamic document and will be updated frequently. Please check back for latest projection. This projection is based on unofficial Bangladesh incidence data.

### Methodology

This projection is based on an SEIR model. Here, the **S** stands for Susceptible, **E** stands for Exposed/infected but asymptomatic, **I** stands for Infected **and** symptomatic, and **R** stands for Recovered. N is the population size.

Assuming there is no births or deaths in a population, (known as a closed population), the model is formulated by the following differential equations.


$$
\begin{align}
\frac{\partial S}{\partial t} & = -\frac{\beta I S}{N} \\
\frac{\partial E}{\partial t} &= \frac{\beta I S}{N} -\sigma E \\
\frac{\partial I}{\partial t} &= \sigma E - \gamma I \\
\frac{\partial R}{\partial t} &= \gamma I
\end{align}
$$

Here the parameters $\beta$ controls the transmission rate, which is the product of contact rate and the probability of transmission given contact betwen S and E compartments. $\sigma$ controls the transition from E to I, and $\gamma$ controls the transition from I to R.

The reproduction rate, $R_0$ can be approximated by 
$$R_0 = \frac{\beta}{\gamma}$$
In plain language, $R_0$ tells us how many people are infected from one patient. An $R_0>1$ indicates the epidemic is at the grotwh phase. $R_0<1$ means the epidemic is slowing or decaying. 

Model was fitted using all but the last two day's incidences to obtain the estimated $\beta$ and $\gamma$. The fitted model was used for prediction. This post was inspired by @churches2020covid19rpart1, where you will find some details on the computation.


### Ascertainment Rate

Not all the cases are reported or tested. Usually a fraction of the actual cases are detected. This is known as ascertainment rate. We consider 25\%, 50\%, 75\% amd 90\% ascertainment rate when fitting the model. 

Simply, the incidences are inflated by the inverse of the ascertainment rate.

Please let me know if you find any error in it. The code was adapted from @churches2020covid19rpart1

```{r, projection-bangladesh-fn,  echo = TRUE, warning=FALSE}

library(deSolve)
library(grid)
library(gridExtra)


######################################
## SIER Modeling -------
######################################
# Parameters
# beta = rate of expusore from susceptible infected contact 
# sigma = rate at which exposed person becomes infected
# gamma = rate at which infected person recovers
# S = Initial susceptible population
# E = Initial exposed population
# I = Initial infected population
# R = Recovered population

fit_seir <- function(country_name='Bangladesh(unoff)', 
                     N=170000000, 
                     af=0.5, npast=2, nfuture=5, 
                     run_date = today()){
  # country = Country name
  # N = population size of the country
  # af = ascertainment factor, default = 0.5
  # country = "Bangladesh(unoff)"
  # npast = number of days in the past to exclude when fitting the model
  # default is npast = 2
  # nfuture = number of days in the future the algorithm to predict to
  # default is nfuture=5
  # run_date = sets the cutoff date so that the later runs do not overrite
  # previous runs. Default is today()
  
  SEIR <- function(time, state, parameters) {
    par <- as.list(c(state, parameters))
    with(par, {
      dS <- -beta * I * S/N
      dE <- beta * I * S/N - sigma * E
      dI <- sigma * E - gamma * I
      dR <- gamma * I
      list(c(dS, dE, dI, dR))
    })
  }

  # define a function to calculate the residual sum of squares
  # (RSS), passing in parameters beta and gamma that are to be
  # optimised for the best fit to the incidence data
  
  RSS <- function(parameters) {
    names(parameters) <- c("beta", "sigma", "gamma")
    out <- ode(y = init, times = Day, func = SEIR, parms = parameters)
    fit <- out[, 4]
    sum((infected - fit)^2)
  }
  
  country = enquo(country_name)
  
  df <- bd_unoff %>% filter(country == !!country, cum_cases>0, date <= run_date)
  infected <- df %>% filter(date >= min(date), date <= today() - 1 - npast) %>% 
    pull(cum_cases)
  
  R = 0; E=0; I = infected[1]; S = N - E - I - R
  
  seir_start_date <- df %>% pull(date) %>% min()
  
  # Ascertainment factor
  infected = infected * 1/af
  
  # Create an incrementing Day vector the same length as our
  # cases vector
  Day <- 1:(length(infected))
  
  # now specify initial values for S, I and R
  init <- c(S = S, E=E, I=I, R=R)
  
  # now find the values of beta and gamma that give the
  # smallest RSS, which represents the best fit to the data.
  # Start with values of 0.5 for each, and constrain them to
  # the interval 0 to 1.0
  
  opt <- optim(c(.5, .5, .5), RSS, method = "L-BFGS-B", 
               lower = c(0.01,0.01,0.01), upper = c(.999, .999, .999), 
               control=list(maxit = 1000))
  
  # check for convergence
  opt_msg = opt$message
  opt_par <- setNames(opt$par, c("beta", "sigma", "gamma"))

  beta = opt_par["beta"]
  gamma = opt_par["gamma"]
  sigma = opt_par["sigma"]
  R0 = as.numeric(beta/gamma)
  
  
  # time in days for predictions
  t <- 1:(as.integer(today() - seir_start_date)  + nfuture)
  
  # get the fitted values from our SEIR model
  
  odefit = ode(y = init, times = t, func = SEIR, parms = opt_par)
  fitted_cases <- data.frame(odefit)
  
  # add a Date column and join the observed incidence data
  fitted_cases <- fitted_cases %>% 
    mutate(date = seir_start_date + days(t - 1)) %>% 
    left_join(df %>% filter(cum_cases>0) %>% ungroup() %>%
                select(date, cum_cases))
  
  # Return
  list(country=country_name, infected = infected,
       opt_msg=opt_msg, opt_par = opt_par, R0=R0, opt_msg=opt_msg, 
       fitted_cases=fitted_cases, N=N, af=af)
  
}

```


```{r, projection-bangladesh-graph, fig.width=10, fig.height=8, echo = FALSE, warning=FALSE, cache=TRUE, message=FALSE}

a1 <- fit_seir(country = "Bangladesh(unoff)",  af=.25, nfuture = 5)
p1<- a1$fitted_cases %>% 
  ggplot(aes(x = date)) + 
  geom_line(aes(y = I), colour = "red") +
  geom_line(aes(y = E), colour = "blue") +
  geom_line(aes(y = R), colour = "darkgreen") +
  geom_point(aes(y = cum_cases), colour = "orange") + 
  scale_y_continuous(labels = scales::comma) + 
  labs(y = "Cumulative incidence", 
       title = paste("R0 =", format(a1$R0, digit=3),"and","assumsed reporting =", a1$af*100,"%" ))

a2 <- fit_seir(country = "Bangladesh(unoff)", af=.5, nfuture = 5)
p2<- a2$fitted_cases %>% 
  ggplot(aes(x = date)) + 
  geom_line(aes(y = I), colour = "red") +
  geom_line(aes(y = E), colour = "blue") +
  geom_line(aes(y = R), colour = "darkgreen") +
  geom_point(aes(y = cum_cases), colour = "orange") + 
  scale_y_continuous(labels = scales::comma) + 
  labs(y = "Cumulative incidence", 
       title = paste("R0 =", format(a2$R0, digit=3),"and","assumsed reporting =", a2$af*100,"%" ))

a3 <- fit_seir(country = "Bangladesh(unoff)", af=.75, nfuture = 5)
p3<- a3$fitted_cases %>% 
  ggplot(aes(x = date)) + 
  geom_line(aes(y = I), colour = "red") +
  geom_line(aes(y = E), colour = "blue") +
  geom_line(aes(y = R), colour = "darkgreen") +
  geom_point(aes(y = cum_cases), colour = "orange") + 
  scale_y_continuous(labels = scales::comma) + 
  labs(y = "Cumulative incidence", 
       title = paste("R0 =", format(a3$R0, digit=3),"and","assumsed reporting =", a3$af*100,"%" ))


a4 <- fit_seir(country = "Bangladesh(unoff)", af=.9, nfuture = 5)
p4<- a4$fitted_cases %>% 
  ggplot(aes(x = date)) + 
  geom_line(aes(y = I), colour = "red") +
  geom_line(aes(y = E), colour = "blue") +
  geom_line(aes(y = R), colour = "darkgreen") +
  geom_point(aes(y = cum_cases), colour = "orange") + 
  scale_y_continuous(labels = scales::comma) + 
  labs(y = "Cumulative incidence", 
       title = paste("R0 =", format(a4$R0, digit=3),"and","assumsed reporting =", a4$af*100,"%" ))



grid.arrange(grobs = list(p1, p2, p3, p4), ncol=2, 
             top = textGrob(paste0("COVID-19 observed and 5-day projected incidence for Bangladesh (unofficial)\n(orange=actual cases, red=predicted incidence from SEIR model, blue=Exposed, green=Recovered)\n",  today()),
             gp=gpar(fontsize=16, font=16)))
```

It turns out that the bottom left one fits the current data best. So lets put that figure on a bigger canvas.


```{r, projection-bangladesh-graph-best, fig.width=10, fig.height=8, echo = FALSE, warning=FALSE}
# Plotting the best fit
p3
```

### Projection for the next 5 days

```{r, echo=FALSE, fig.cap="Actual and Predicted Cumulative Cases using SEIR Model"}
a3$fitted_cases %>% as.data.frame() %>%
  mutate(
    Date = date,
    `Actual daily cases` = c(0, diff(cum_cases)),
    `Actual cumulative cases` = cum_cases,
    `Projected cumulative cases` = round(I, digits = 0),
    `Projected daily cases` = c(0, diff(`Projected cumulative cases`))
  ) %>%
  select(Date, `Actual daily cases`, `Projected daily cases`,
         `Actual cumulative cases`,
          `Projected cumulative cases` ) %>%
  tail(10) %>%
  kable(format = "html", 
        caption=paste("Predicted new cases for the next 5 days")) %>%
          kable_styling(position = "center")
```

### Projection for 100 days into the future

Assuming the situation will remain like this including the interventions currently in place, the 100 day projection suggests that the the peak of the epidemic will be around the middle of June. The trajectory also suggests that the epidemic will end by end of July or early August. 

```{r, projection-bangladesh-graph-100, fig.width=10, fig.height=8, echo = FALSE, warning=FALSE, cache=TRUE, message=FALSE}

nfuture=100

a1 <- fit_seir(country = "Bangladesh(unoff)", af=.25, nfuture = nfuture)
p1<- a1$fitted_cases %>% 
  ggplot(aes(x = date)) + 
  geom_line(aes(y = I), colour = "red") +
  geom_line(aes(y = E), colour = "blue") +
  # geom_line(aes(y = R), colour = "darkgreen") +
  geom_point(aes(y = cum_cases), colour = "orange") + 
  scale_y_continuous(labels = scales::comma) + 
  labs(y = "Cumulative incidence", 
       title = paste("R0 =", format(a1$R0, digit=3),"and","assumsed reporting =", a1$af*100,"%" ))

a2 <- fit_seir(country = "Bangladesh(unoff)", af=.5, nfuture = nfuture)
p2<- a2$fitted_cases %>% 
  ggplot(aes(x = date)) + 
  geom_line(aes(y = I), colour = "red") +
  geom_line(aes(y = E), colour = "blue") +
  # geom_line(aes(y = R), colour = "darkgreen") +
  geom_point(aes(y = cum_cases), colour = "orange") + 
  scale_y_continuous(labels = scales::comma) + 
  labs(y = "Cumulative incidence", 
       title = paste("R0 =", format(a2$R0, digit=3),"and","assumsed reporting =", a2$af*100,"%" ))

a3 <- fit_seir(country = "Bangladesh(unoff)", af=.75, nfuture = nfuture)
p3<- a3$fitted_cases %>% 
  ggplot(aes(x = date)) + 
  geom_line(aes(y = I), colour = "red") +
  geom_line(aes(y = E), colour = "blue") +
  # geom_line(aes(y = R), colour = "darkgreen") +
  geom_point(aes(y = cum_cases), colour = "orange") + 
  scale_y_continuous(labels = scales::comma) + 
  labs(y = "Cumulative incidence", 
       title = paste("R0 =", format(a3$R0, digit=3),"and","assumsed reporting =", a3$af*100,"%" ))


a4 <- fit_seir(country = "Bangladesh(unoff)", af=.9, nfuture = nfuture)
p4<- a4$fitted_cases %>% 
  ggplot(aes(x = date)) + 
  geom_line(aes(y = I), colour = "red") +
  geom_line(aes(y = E), colour = "blue") +
  # geom_line(aes(y = R), colour = "darkgreen") +
  geom_point(aes(y = cum_cases), colour = "orange") + 
  scale_y_continuous(labels = scales::comma) +
  # scale_y_continuous(labels = scales::comma, trans = 'log') + 
  labs(y = "Cumulative incidence", 
       title = paste("R0 =", format(a4$R0, digit=3),"and","assumsed reporting =", a4$af*100,"%" ))


grid.arrange(grobs = list(p1, p2, p3, p4), ncol=2, 
             top = textGrob(paste0("COVID-19 observed and 100-day projected  incidence for Bangladesh (unofficial)\n(orange=actual cases, red=predicted incidence from SEIR model, blue=Exposed)\n",  today()),
             gp=gpar(fontsize=16, font=16)))

# grid.arrange(a, b, c, d,ncol=2, 
#      nrow=2, top=textGrob("Title", gp=gpar(fontsize=15,font=8)))

```


## Projection for Bangladesh (unofficial) Run 2 {#projection-bangladesh-run2}

__Run date__: 21 April 2020


This is the second projection. For the initial project, please scroll up. The methodology is discussed above.


```{r, projection-bangladesh-graph-run2, fig.width=10, fig.height=8, echo = FALSE, warning=FALSE, cache=TRUE, message=FALSE}

run_date = ymd("2020-04-21")
a1 <- fit_seir(country = "Bangladesh(unoff)",  af=.25, nfuture = 5, 
               run_date = run_date)
p1<- a1$fitted_cases %>% 
  ggplot(aes(x = date)) + 
  geom_line(aes(y = I), colour = "red") +
  geom_line(aes(y = E), colour = "blue") +
  geom_line(aes(y = R), colour = "darkgreen") +
  geom_point(aes(y = cum_cases), colour = "orange") + 
  scale_y_continuous(labels = scales::comma) + 
  labs(y = "Cumulative incidence", 
       title = paste("R0 =", format(a1$R0, digit=3),"and","assumsed reporting =", a1$af*100,"%" ))

a2 <- fit_seir(country = "Bangladesh(unoff)", af=.5, nfuture = 5, 
               run_date = run_date)
p2<- a2$fitted_cases %>% 
  ggplot(aes(x = date)) + 
  geom_line(aes(y = I), colour = "red") +
  geom_line(aes(y = E), colour = "blue") +
  geom_line(aes(y = R), colour = "darkgreen") +
  geom_point(aes(y = cum_cases), colour = "orange") + 
  scale_y_continuous(labels = scales::comma) + 
  labs(y = "Cumulative incidence", 
       title = paste("R0 =", format(a2$R0, digit=3),"and","assumsed reporting =", a2$af*100,"%" ))

a3 <- fit_seir(country = "Bangladesh(unoff)", af=.75, nfuture = 5,
               run_date = run_date)
p3<- a3$fitted_cases %>% 
  ggplot(aes(x = date)) + 
  geom_line(aes(y = I), colour = "red") +
  geom_line(aes(y = E), colour = "blue") +
  geom_line(aes(y = R), colour = "darkgreen") +
  geom_point(aes(y = cum_cases), colour = "orange") + 
  scale_y_continuous(labels = scales::comma) + 
  labs(y = "Cumulative incidence", 
       title = paste("R0 =", format(a3$R0, digit=3),"and","assumsed reporting =", a3$af*100,"%" ))


a4 <- fit_seir(country = "Bangladesh(unoff)", af=.9, nfuture = 5, run_date = run_date)
p4<- a4$fitted_cases %>% 
  ggplot(aes(x = date)) + 
  geom_line(aes(y = I), colour = "red") +
  geom_line(aes(y = E), colour = "blue") +
  geom_line(aes(y = R), colour = "darkgreen") +
  geom_point(aes(y = cum_cases), colour = "orange") + 
  scale_y_continuous(labels = scales::comma) + 
  labs(y = "Cumulative incidence", 
       title = paste("R0 =", format(a4$R0, digit=3),"and","assumsed reporting =", a4$af*100,"%" ))



grid.arrange(grobs = list(p1, p2, p3, p4), ncol=2, 
             top = textGrob(paste0("COVID-19 observed and 5-day projected incidence for Bangladesh (unofficial)\n(orange=actual cases, red=predicted incidence from SEIR model, blue=Exposed, green=Recovered)\n",  today()),
             gp=gpar(fontsize=16, font=16)))
```

It turns out that the bottom right one fits the current data best. So lets put that figure on a bigger canvas. Please note that in Run 1, we used a similar $R_0$ but a lower ascertainment rate. This time, a higher ascertainment rate is associated with a similar $R_0$.


```{r, projection-bangladesh-graph-best-run2, fig.width=10, fig.height=8, echo = FALSE, warning=FALSE}
# Plotting the best fit
# p4 + ggtitle(paste0("Best fit model\n(orange=actual cases, red=predicted incidence from SEIR model, blue=Exposed, green=Recovered)", " ", run_date))

p4 + ggtitle(paste0("Best fit model as of ", run_date, "\n(orange=actual cases, red=predicted incidence from SEIR model, blue=Exposed, green=Recovered)\n", " ","R0 =", format(a4$R0, digit=3)," and ", "assumsed reporting= ", a4$af*100,"%"))
       

```

### Projection for the next 5 days

```{r, echo=FALSE, fig.cap="Actual and Predicted Cumulative Cases using SEIR Model"}
a4$fitted_cases %>% as.data.frame() %>%
  mutate(
    Date = date,
    `Actual daily cases` = c(0, diff(cum_cases)),
    `Actual cumulative cases` = cum_cases,
    `Projected cumulative cases` = round(I, digits = 0),
    `Projected daily cases` = c(0, diff(`Projected cumulative cases`))
  ) %>%
  select(Date, `Actual daily cases`, `Projected daily cases`,
         `Actual cumulative cases`,
          `Projected cumulative cases` ) %>%
  tail(12) %>%
  kable(format = "html", 
        caption=paste("Predicted new cases for the next 5 days")) %>%
          kable_styling(position = "center")
```

### Projection for 100 days into the future

Assuming the situation will remain like this including the interventions currently in place, the 100 day projection suggests that the the peak of the epidemic will be around the middle of June. The trajectory also suggests that the epidemic will end by end of July or early August. 

```{r, projection-bangladesh-graph-100-run2, fig.width=10, fig.height=8, echo = FALSE, warning=FALSE, cache=TRUE, message=FALSE}

nfuture=100

a1 <- fit_seir(country = "Bangladesh(unoff)", af=.25, nfuture = nfuture, run_date = run_date)
p1<- a1$fitted_cases %>% 
  ggplot(aes(x = date)) + 
  geom_line(aes(y = I), colour = "red") +
  geom_line(aes(y = E), colour = "blue") +
  # geom_line(aes(y = R), colour = "darkgreen") +
  geom_point(aes(y = cum_cases), colour = "orange") + 
  scale_y_continuous(labels = scales::comma) + 
  labs(y = "Cumulative incidence", 
       title = paste("R0 =", format(a1$R0, digit=3),"and","assumsed reporting =", a1$af*100,"%" ))

a2 <- fit_seir(country = "Bangladesh(unoff)", af=.5, nfuture = nfuture)
p2<- a2$fitted_cases %>% 
  ggplot(aes(x = date)) + 
  geom_line(aes(y = I), colour = "red") +
  geom_line(aes(y = E), colour = "blue") +
  # geom_line(aes(y = R), colour = "darkgreen") +
  geom_point(aes(y = cum_cases), colour = "orange") + 
  scale_y_continuous(labels = scales::comma) + 
  labs(y = "Cumulative incidence", 
       title = paste("R0 =", format(a2$R0, digit=3),"and","assumsed reporting =", a2$af*100,"%" ))

a3 <- fit_seir(country = "Bangladesh(unoff)", af=.75, nfuture = nfuture)
p3<- a3$fitted_cases %>% 
  ggplot(aes(x = date)) + 
  geom_line(aes(y = I), colour = "red") +
  geom_line(aes(y = E), colour = "blue") +
  # geom_line(aes(y = R), colour = "darkgreen") +
  geom_point(aes(y = cum_cases), colour = "orange") + 
  scale_y_continuous(labels = scales::comma) + 
  labs(y = "Cumulative incidence", 
       title = paste("R0 =", format(a3$R0, digit=3),"and","assumsed reporting =", a3$af*100,"%" ))


a4 <- fit_seir(country = "Bangladesh(unoff)", af=.9, nfuture = nfuture)
p4<- a4$fitted_cases %>% 
  ggplot(aes(x = date)) + 
  geom_line(aes(y = I), colour = "red") +
  geom_line(aes(y = E), colour = "blue") +
  # geom_line(aes(y = R), colour = "darkgreen") +
  geom_point(aes(y = cum_cases), colour = "orange") + 
  scale_y_continuous(labels = scales::comma) +
  # scale_y_continuous(labels = scales::comma, trans = 'log') + 
  labs(y = "Cumulative incidence", 
       title = paste("R0 =", format(a4$R0, digit=3),"and","assumsed reporting =", a4$af*100,"%" ))


grid.arrange(grobs = list(p1, p2, p3, p4), ncol=2, 
             top = textGrob(paste0("COVID-19 observed and 100-day projected  incidence for Bangladesh (unofficial)\n(orange=actual cases, red=predicted incidence from SEIR model, blue=Exposed)\n",  today()),
             gp=gpar(fontsize=16, font=16)))

# grid.arrange(a, b, c, d,ncol=2, 
#      nrow=2, top=textGrob("Title", gp=gpar(fontsize=15,font=8)))

```



<!-- ## Projection for Bangladesh (unofficial) Run 3 {#projection-bangladesh-run3} -->

<!-- __Run date__: 11 May 2020 -->


<!-- This is the third projection. For the initial project, please scroll up. The methodology is discussed above. -->


<!-- ```{r, projection-bangladesh-graph-run3, fig.width=10, fig.height=8, echo = FALSE, warning=FALSE, cache=TRUE, message=FALSE} -->

<!-- run_date = ymd("2020-05-11") -->
<!-- a1 <- fit_seir(country = "Bangladesh(unoff)",  af=.25, nfuture = 5, -->
<!--                run_date = run_date) -->
<!-- p1<- a1$fitted_cases %>% -->
<!--   ggplot(aes(x = date)) + -->
<!--   geom_line(aes(y = I), colour = "red") + -->
<!--   geom_line(aes(y = E), colour = "blue") + -->
<!--   geom_line(aes(y = R), colour = "darkgreen") + -->
<!--   geom_point(aes(y = cum_cases), colour = "orange") + -->
<!--   scale_y_continuous(labels = scales::comma) + -->
<!--   labs(y = "Cumulative incidence", -->
<!--        title = paste("R0 =", format(a1$R0, digit=3),"and","assumsed reporting =", a1$af*100,"%" )) -->

<!-- a2 <- fit_seir(country = "Bangladesh(unoff)", af=.5, nfuture = 5, -->
<!--                run_date = run_date) -->
<!-- p2<- a2$fitted_cases %>% -->
<!--   ggplot(aes(x = date)) + -->
<!--   geom_line(aes(y = I), colour = "red") + -->
<!--   geom_line(aes(y = E), colour = "blue") + -->
<!--   geom_line(aes(y = R), colour = "darkgreen") + -->
<!--   geom_point(aes(y = cum_cases), colour = "orange") + -->
<!--   scale_y_continuous(labels = scales::comma) + -->
<!--   labs(y = "Cumulative incidence", -->
<!--        title = paste("R0 =", format(a2$R0, digit=3),"and","assumsed reporting =", a2$af*100,"%" )) -->

<!-- a3 <- fit_seir(country = "Bangladesh(unoff)", af=.75, nfuture = 5, -->
<!--                run_date = run_date) -->
<!-- p3<- a3$fitted_cases %>% -->
<!--   ggplot(aes(x = date)) + -->
<!--   geom_line(aes(y = I), colour = "red") + -->
<!--   geom_line(aes(y = E), colour = "blue") + -->
<!--   geom_line(aes(y = R), colour = "darkgreen") + -->
<!--   geom_point(aes(y = cum_cases), colour = "orange") + -->
<!--   scale_y_continuous(labels = scales::comma) + -->
<!--   labs(y = "Cumulative incidence", -->
<!--        title = paste("R0 =", format(a3$R0, digit=3),"and","assumsed reporting =", a3$af*100,"%" )) -->


<!-- a4 <- fit_seir(country = "Bangladesh(unoff)", af=.9, nfuture = 5, run_date = run_date) -->
<!-- p4<- a4$fitted_cases %>% -->
<!--   ggplot(aes(x = date)) + -->
<!--   geom_line(aes(y = I), colour = "red") + -->
<!--   geom_line(aes(y = E), colour = "blue") + -->
<!--   geom_line(aes(y = R), colour = "darkgreen") + -->
<!--   geom_point(aes(y = cum_cases), colour = "orange") + -->
<!--   scale_y_continuous(labels = scales::comma) + -->
<!--   labs(y = "Cumulative incidence", -->
<!--        title = paste("R0 =", format(a4$R0, digit=3),"and","assumsed reporting =", a4$af*100,"%" )) -->



<!-- grid.arrange(grobs = list(p1, p2, p3, p4), ncol=2, -->
<!--              top = textGrob(paste0("COVID-19 observed and 5-day projected incidence for Bangladesh (unofficial)\n(orange=actual cases, red=predicted incidence from SEIR model, blue=Exposed, green=Recovered)\n",  today()), -->
<!--              gp=gpar(fontsize=16, font=16))) -->
<!-- ``` -->

<!-- It turns out that the bottom right one fits the current data best. So lets put that figure on a bigger canvas. Please note that in Run 1, we used a similar $R_0$ but a lower ascertainment rate. This time, a higher ascertainment rate is associated with a similar $R_0$. -->


<!-- ```{r, projection-bangladesh-graph-best-run3, fig.width=10, fig.height=8, echo = FALSE, warning=FALSE} -->
<!-- # Plotting the best fit -->
<!-- # p4 + ggtitle(paste0("Best fit model\n(orange=actual cases, red=predicted incidence from SEIR model, blue=Exposed, green=Recovered)", " ", run_date)) -->

<!-- p4 + ggtitle(paste0("Best fit model as of ", run_date, "\n(orange=actual cases, red=predicted incidence from SEIR model, blue=Exposed, green=Recovered)\n", " ","R0 =", format(a4$R0, digit=3)," and ", "assumsed reporting= ", a4$af*100,"%")) -->



<!-- a5 <- fit_seir(country = "Bangladesh(unoff)", af=.96, npast = 0, nfuture = 5, run_date = run_date) -->
<!-- p5<- a5$fitted_cases %>% -->
<!--   ggplot(aes(x = date)) + -->
<!--   geom_line(aes(y = I), colour = "red") + -->
<!--   geom_line(aes(y = E), colour = "blue") + -->
<!--   geom_line(aes(y = R), colour = "darkgreen") + -->
<!--   geom_point(aes(y = cum_cases), colour = "orange") + -->
<!--   scale_y_continuous(labels = scales::comma) + -->
<!--   labs(y = "Cumulative incidence", -->
<!--        title = paste("R0 =", format(a5$R0, digit=3),"and","assumsed reporting =", a5$af*100,"%" )) -->

<!-- p5 -->

<!-- ``` -->

<!-- ### Projection for the next 5 days -->

<!-- ```{r, echo=FALSE, fig.cap="Actual and Predicted Cumulative Cases using SEIR Model"} -->

<!-- a5 <- fit_seir(country = "Bangladesh(unoff)", af=1.1, npast = 0, nfuture = 5, run_date = run_date) -->
<!-- a5$fitted_cases %>% as.data.frame() %>% -->
<!--   mutate( -->
<!--     Date = date, -->
<!--     `Actual daily cases` = c(0, diff(cum_cases)), -->
<!--     `Actual cumulative cases` = cum_cases, -->
<!--     `Projected cumulative cases` = round(I, digits = 0), -->
<!--     `Projected daily cases` = c(0, diff(`Projected cumulative cases`)) -->
<!--   ) %>% -->
<!--   select(Date, `Actual daily cases`, `Projected daily cases`, -->
<!--          `Actual cumulative cases`, -->
<!--           `Projected cumulative cases` ) %>% -->
<!--   tail(12) %>% -->
<!--   kable(format = "html", -->
<!--         caption=paste("Predicted new cases for the next 5 days")) %>% -->
<!--           kable_styling(position = "center") -->
<!-- ``` -->

<!-- ### Projection for 100 days into the future -->

<!-- Assuming the situation will remain like this including the interventions currently in place, the 100 day projection suggests that the the peak of the epidemic will be around the middle of June. The trajectory also suggests that the epidemic will end by end of July or early August. -->

<!-- ```{r, projection-bangladesh-graph-100-run3, fig.width=10, fig.height=8, echo = FALSE, warning=FALSE, cache=TRUE, message=FALSE} -->

<!-- nfuture=100 -->

<!-- a1 <- fit_seir(country = "Bangladesh(unoff)", af=.25, nfuture = nfuture, run_date = run_date) -->
<!-- p1<- a1$fitted_cases %>% -->
<!--   ggplot(aes(x = date)) + -->
<!--   geom_line(aes(y = I), colour = "red") + -->
<!--   geom_line(aes(y = E), colour = "blue") + -->
<!--   # geom_line(aes(y = R), colour = "darkgreen") + -->
<!--   geom_point(aes(y = cum_cases), colour = "orange") + -->
<!--   scale_y_continuous(labels = scales::comma) + -->
<!--   labs(y = "Cumulative incidence", -->
<!--        title = paste("R0 =", format(a1$R0, digit=3),"and","assumsed reporting =", a1$af*100,"%" )) -->

<!-- a2 <- fit_seir(country = "Bangladesh(unoff)", af=.5, nfuture = nfuture) -->
<!-- p2<- a2$fitted_cases %>% -->
<!--   ggplot(aes(x = date)) + -->
<!--   geom_line(aes(y = I), colour = "red") + -->
<!--   geom_line(aes(y = E), colour = "blue") + -->
<!--   # geom_line(aes(y = R), colour = "darkgreen") + -->
<!--   geom_point(aes(y = cum_cases), colour = "orange") + -->
<!--   scale_y_continuous(labels = scales::comma) + -->
<!--   labs(y = "Cumulative incidence", -->
<!--        title = paste("R0 =", format(a2$R0, digit=3),"and","assumsed reporting =", a2$af*100,"%" )) -->

<!-- a3 <- fit_seir(country = "Bangladesh(unoff)", af=.75, nfuture = nfuture) -->
<!-- p3<- a3$fitted_cases %>% -->
<!--   ggplot(aes(x = date)) + -->
<!--   geom_line(aes(y = I), colour = "red") + -->
<!--   geom_line(aes(y = E), colour = "blue") + -->
<!--   # geom_line(aes(y = R), colour = "darkgreen") + -->
<!--   geom_point(aes(y = cum_cases), colour = "orange") + -->
<!--   scale_y_continuous(labels = scales::comma) + -->
<!--   labs(y = "Cumulative incidence", -->
<!--        title = paste("R0 =", format(a3$R0, digit=3),"and","assumsed reporting =", a3$af*100,"%" )) -->


<!-- a4 <- fit_seir(country = "Bangladesh(unoff)", af=1.1, nfuture = nfuture) -->
<!-- p4<- a4$fitted_cases %>% -->
<!--   ggplot(aes(x = date)) + -->
<!--   geom_line(aes(y = I), colour = "red") + -->
<!--   geom_line(aes(y = E), colour = "blue") + -->
<!--   # geom_line(aes(y = R), colour = "darkgreen") + -->
<!--   geom_point(aes(y = cum_cases), colour = "orange") + -->
<!--   scale_y_continuous(labels = scales::comma) + -->
<!--   # scale_y_continuous(labels = scales::comma, trans = 'log') + -->
<!--   labs(y = "Cumulative incidence", -->
<!--        title = paste("R0 =", format(a4$R0, digit=3),"and","assumsed reporting =", a4$af*100,"%" )) -->


<!-- grid.arrange(grobs = list(p1, p2, p3, p4), ncol=2, -->
<!--              top = textGrob(paste0("COVID-19 observed and 100-day projected  incidence for Bangladesh (unofficial)\n(orange=actual cases, red=predicted incidence from SEIR model, blue=Exposed)\n",  today()), -->
<!--              gp=gpar(fontsize=16, font=16))) -->

<!-- # grid.arrange(a, b, c, d,ncol=2, -->
<!-- #      nrow=2, top=textGrob("Title", gp=gpar(fontsize=15,font=8))) -->

<!-- ``` -->


<!--chapter:end:04-projections.Rmd-->


# Publications {#scipubs}

This section lists major publications with a brief summary. The latest publications are listed first. 

- Estimates of the severity of coronavirus disease 2019: a model-based analysis. https://doi.org/10.1016/S1473-3099(20)30243-7

  Using data on 24 deaths that occurred in mainland China and 165 recoveries outside of China, we estimated the mean duration from onset of symptoms to death to be 17·8 days (95% credible interval [CrI] 16·9–19·2) and to hospital discharge to be 24·7 days (22·9–28·1). In all laboratory confirmed and clinically diagnosed cases from mainland China (n=70 117), we estimated a crude case fatality ratio (adjusted for censoring) of 3·67% (95% CrI 3·56–3·80). However, after further adjusting for demography and under-ascertainment, we obtained a best estimate of the case fatality ratio in China of 1·38% (1·23–1·53), with substantially higher ratios in older age groups (0·32% [0·27–0·38] in those aged <60 years vs 6·4% [5·7–7·2] in those aged ≥60 years), up to 13·4% (11·2–15·9) in those aged 80 years or older. 
 
## Other articles

- [Why COVID19 can't be compared with flu by Ruth Etzioni](https://timmermanreport.com/2020/04/why-covid-19-cant-be-directly-compared-with-the-flu/) 

<!--chapter:end:05-publications.Rmd-->


# Stories

__Report last updated:__ `r Sys.Date()`


## Rumi Ahmed

### April 7, 2020 (via Facebook)

I lived through the swine flu (2009 H1N1) pandemic which infected 60 million (1 in 6 American) and killed about 12000 Americans. Global death toll was around 300000.

Then we lived through Bird Flu (50% mortality), Ebola (70% mortality), MERS(40% mortality), Zika (airborne and. It this defect) scare and cattle call.

So when COVID-19 started ravaging Wuhan, China and Lombardy, Italy - me and many of my frontline, more informed physician colleague were like the mythical shepherd boys. We Have been dismissive!

Dismissive because, with R0 of 2.5-2.9, SARS-COV-2 was less contagious than SARS-1 of 2002 and definitely much less fatal than SARS-1, MERS, Bird Flu, Ebola etc. We knew that SARS-1, MERS, Avian influenza blazed for a few months in a few localities then stoped abruptly. R0 became <1, the epidemic died down.

And SARS-COV-2 was a petty coronavirus! It was a mere coronavirus!! It was not the feared shifted- drifted influenza virus causing Flu pandemic we have been fearing half of the last century.

Despite SARS -1 and MERS experience - we didn’t give this bat coronavirus due respects it deserved. It was not the feared white walker, it was not the novel influenza pandemic we always have been fearful about.

But this novel coronavirus SARA-COV-2 turned out to be the cause of first coronavirus pandemic, to be more precise , the first non-influenza pandemic in history of mankind.

So what went wrong with this Novel coronavirus? Biologically less contagious and less fatal than SARS-1, instead of behaving like SARS-1 or MERS, why this novel coronavirus is blazing like a catastrophic forest fire?

The answer is in one protein. Remember this name. It’s a protein called Exonuclease.
Exonuclease!

We are still inside the deep fog of war, at the center of a bloody- muddy battle field, we are wielding swords blindly! We don’t have the clarity of a bird eye view or the luxury of a time lapse retrospect! Hazy view through early data says the following about this new enemy - this virus-

1. It is a bat virus, and thorough perpetual mixing and mutation inside the bat reservoir, this bat virus mutated and developed as a new virus sometimes between 1948 and 1972. Chinese horseshoe bats are carrying this virus for the last several decades!
2. Like tornado chasers - there are a group of scientists knows as virus chasers who are skimming wild bat habitants in search of viruses that they call pre-pandemic viruses. They are isolating bat viruses which in the future has the potential to transmit to human and cause a new pandemic. They are pre-preparing for the next pandemic by sequencing them and trying to see vaccine potential. This new SARS-COV-2 was not in that pre-pandemic bat virus database. In fact this virus was never known to us despite it was circulating inside horseshoe bats for the last several decades. It says that our limited search efforts with lackluster funding was not enough to capture many bat viruses.
3. This horseshoe Chinese cave bats are not fruit bats- they have teeth and may infect human / other animals by any close contact.
4. Somehow humans got this SARS-COV-2 bat virus and stated getting sick. Did this jump ( virus changing host from bat to human) happen last fall? Did similar COVID-19 disease clusters happen before 2019 but we didn’t notice? Very possible.
5. Remember AIDS. We always thought that HIV virus came to Human from Chimpanzee in 1983. But we now know that the jump probably happened in 1929! Since 1929, there have been undetected local epidemics AIDS.
6. So SARS-COV-2 virus found this new host in human species which is abundant in planet earth.
7. This virus, in this new host, has only one job in life- propagate!
8. Remember the evolutionary theory of selection. This virus came this far- and so far winning against the most powerful species ever lived in this planet, Homo sapiens - this virus is the fittest. Genetically and biochemically, this virus is the chosen one, the selected one to survive and propagate. Millions / billions of other weaker viral species got defeated / lost out - this one didn’t! This is pure simple theory of selection!

Why this virus has been selected to come this far and win over a complex new host? There are many factors and explanations! But one factor is that it has a protein called Exonuclease in its arsenal!

Exonuclease is a protein this virus generates for a job called genetic proof-reading! Viruses are constantly replicating into new viruses. That’s how they propagate. Usually viral spreads die down after an initially blaze because the progeny keeps getting weaker and weaker due to constant mutations. Usually viruses don’t have a quality control officer in its payroll!

But this virus has a powerful quality control officer. This Exonuclease protein is ensuring that every single progeny this virus is producing is a healthy progeny. It’s not allowing heavily mutated viruses which usually terminated the chain of transmission. And because of this Exonuclease, unlike SARS-1, MERS, this virus is not suddenly disappearing from the horizon; rather with Alexander the Great vigor, this virus is marching - conquering country after country, sacking city after city!


### April 16, 2020 - Bats (via Facebook)

Of bats and viruses!

Part -1 Bats.

I am jealous of bats.

Bat is a mammal, I am also a mammal; and I am definitely a bigger mammal. I can’t fly while bats can! But I am jealous of bats not only because bats can fly I can’t!

There are many more thing about bats that makes us look at this species of mammals with awe! Although we don’t have a clear bat census - bats, as well as rodents, are the two mammals who probably outnumber humans. Of all the mammals in this planet, 20% are bats.

We humans are children compared to bats physical capability. A bat flies 5-6 hours nonstop nightly.
Their maneuvering capacity is even better than birds. When bats fly they can increase their metabolism so high that their heart beats over one thousand times a minute. Compared to that if our heart beats 300 times a minute for a few minute - it stops!

During flying bats body temperature peaks as high as 108F / 42C. A bat body can easily tolerate that high temperature. If that happened in human - the brain would have been fried leading to seizure!

To process all that much metabolism, bats have a pair of super kidneys that’s like 20 pair of human kidneys to remove all these metabolites!

Bats immune system is another nature’s wonder. If bat’s immune system is equivalent to Windows 10, our immune system may be like MS DOS!

Each Bat can carry over hundred types of viruses. Several types of coronavirus including SARS, MERS and current SARS - COV-2 live in bats, as do much more dangerous viruses like Rabies, Ebola, Nipah, Hendra!

See we are doing hand hygiene, social distancing - still we are struggling to safeguard ourselves from virus. Bats neither do hand hygiene nor they follow social distancing as they live in a crowded group setting. Yet they don’t get sick of viruses!

Bat’s immune system can seamlessly handle these viruses for ages, live together not harming each other. But if human gets one of these viruses, they either gets totally defeated in first attack or launch such an erratic immune response - it kills itself by its own action! And exactly the same thing we are seeing in coronavirus pandemic these days!

Bat, as well as rodents are major reservoir of viruses and probably are the source of our current pandemic and probably a more deadly pandemic in the future.

Including three new, there are seven coronavirus in humans. Of these seven, it seems at least six came from to us from bats. Coronavirus 229E jumped from bat to human in ancient times. While the SARS-COV-2 possibly jumped sometime last November!

One human cold coronavirus - HKU3 - it is probably the only one that came to us from cows! It probably happened about a hundred years ago. Although we don’t have a official record - scientists have some evidence that coronavirus HKU3 possibly caused serious illness when first jumped to humans!

If we take history as an indicator for the future - it is more likely that the next COVID-19 crisis will start with another virus from bat.

That’s why our human race will have to invest a lot of our resources in learning more about bats and the viruses that reside in them and the interactions between them!

Current SARS-COV-2 virus is presumed to be a bat virus. But we’ll never be certain about it unless until some day we’ll be able to collect, culture and isolate this virus in a bat.

That’s what Dr. Linfa Wang did. SARS outbreak happened in 2002-2003. We had a genetic sequence suggesting it’s bat origin. But that was not enough. We have to find it in a bat! Dr. Linfa wang of Duke Global Health Institute at Duke National University of Singapore did exactly that thing. He spent ten years time searching / crawling / climbing cave to cave in remote rural China. He would collect bat droplet and real bat sample swab and look for a genetic match! And after ten years search he finally found the SARS virus in a bat!

We talked about the bat lady Dr Shi Zheng-Li of Wuhan institution of Virology that day . She is doing the same thing. As are two agencies in USA, PREDICT -1 and PREDICT -2. Unbeknownst to all of us -there is an active search for bat viruses, the virus that can attack us at some point in future!

— In the second part I’ll discuss more about the viruses. Stay tuned!
Part 2 will be about viruses!



### April 21, 2020 - Bats (Part 2 and 3)

1. Talking about viruses means throwing in a lot of large mathematical values and big numbers - the extent of which most of our minds are unable to fathom.

2. We know quiet a bit about people being infected with coronavirus. We we also see that while some are becoming sick,
a lot more who are invaded / infected by the virus but are not getting sick.

3. Viruses probably doesn’t want to kill us. Killing the new host is not in the best interest for ￼the virus also. These virus are here today after 3 million years of evolution and selection. And they are genetically and proteomically fittest, hence very apt at surviving and spreading in current zoonome.

4. Only thing they want is to make our body their home and live without harming us much. We are great hosts for them - because there are seven billion of us and we live a social life - in close congregation - so if needed they can jump from one to another. We are their dream home!

5. Once they can safely reside in us - we are so many in number - they won’t have to look for new home- their only job will be to keep propagating - for millions of years!

6. Naturally and understandably, our body is home to innumerable viruses. A good count or number is impossible to deduct. We have 40 trillion bacteria in our body and usually we know that there are ten times more viruses in nature than bacteria. Hence it is assumed that there are 400 trillion viruses who call out body their home.

7. We have a peace treaty with these viruses, there is a longstanding ceasefire. We don’t fight each other. Many, if not most, viruses are helpful for us.

8. But how can a virus be helpful? Before we discuss that let’s first talk about virus biodiversity in our body. We are still learning and our knowledge of human virome is still very incomplete.

9. So far what we have identified - there are 53 different types of human viruses in our body and about 53000 types of viruses which are the viruses living inside the bacteria that live inside our body. We know this from gene sequencing study - really isolating them in culture has been an impossible job.

10. Those viruses in bacteria keeps bacteria population in our body under check and don’t let bacteria overwhelm us/ take over us!


11. Humans are mammals. Roughly there are about 6000 species of mammals. How many virus species each mammal carry? So far there has been one real study. A big group of scientists led by world top emerging infection scientist - Dr. Ian Lipkin went to Bangladesh - randomly selected a site - collected flying fox ( fruit bats) for throat swabs, collected urine and stool specimen. A total of -1800 samples! On average they found 58 viruses per bat. Almost all the viruses they identified were new to us. They isolated few bat coronaviruses and most other types of knows virus types. Based on that model we think that mammals have about 400000 different types of viruses, the number rises to 100 million types of we count all vertebrates!

12. But our land based virus is nothing compared to the viruses found in our oceans. Canadian virologist Curtis Suttle always use colorful / sensational allegories when he tries to describe the amount of viruses that reside in our ocean waters.

13. There are 10 to the power 30 number of viruses in our oceans. What does the number 10 to the power 30 mean? It’s a number that normal human brain like mine can’t process. Let me try to explain- if we line up all these marine viruses (100 micron in length on average) side by side that will stretch 10 to the power 28 Kilometers. Believe me that’s a very long distance - about 100 million light years long! Our galaxy is about 150,000 light years across! This line of viruses will span 60 galaxies! How far is the closest star system from earth? Proxima Centauri it is and it’s only about 4.2 light years away!!

14. How much all these ocean virus weight? Their collective weight is equal to the weight of 75 million blue whales!

15. About 10 to power 23 infections are happening every second in our oceans! Virus is the major determinant in the delicate echo balance and marine life cycle and are responsible of 50 % of planet’s oxygen production!

16. In every liter of coastal water there are 10 million viruses. Overall in oceans there are 10 million times more viruses than stars in the galaxy!

17. So that’s about water / sea / river! What about the sky?

Well, about 9000 feet above ground, in the sphere called troposphere, there is transcontinental air current above our weather system! Billions of viral and bacterial particles fly all the way to that high and do transcontinental travel for thousands of miles! And then they shower on us in billions! A single square meter of the planet's surface could be showered with hundreds of millions of viruses — and tens of millions of bacteria — in a single day. These viruses are all over in the air around us. Luckily they are not the types of viruses that are pathogenic to us! Most can’t enter our cells!

18. Measles is a type of virus that may remain air suspended and travel to the far corner of the room staying suspended in air and may infect everyone breathing that air. So are Zoaster! They can fly in air and infect people. As far as we known the current SARS-COV-2 doesn’t have that ability! Luckily Most virus doesn’t have that capacity.

19. Viruses need receptors/ window or door in the cell to enter the cell. Without that they can’t enter. Once virus enters cells - it depends on the Virus how long it will take to start the symptom. Influenza virus symptom starts only on one to two days day after infection! On the other had it may take AIDS virus upto 18 years to start the symptoms!

20. Typically thousands or tens of thousands of coronaviruses need to enter the cells of nose / throats of a person to infect him/ her! But that number is nothing because a hundred million coronavirus particles could fit on the head of a pin. And although we read, worry, talk, discuss about how many days this virus can stay alive on what surface, we need to know that infection window is highest in the first ten minutes.



<!--chapter:end:06-stories.Rmd-->


# Trials and Treatments {#trials}


## Clinical trials


**April 7, 2020**

- [Canada begins clinical trial of experimental COVID-19 treatment using plasma from recovered individuals](https://www.theglobeandmail.com/canada/article-canada-begins-clinical-trial-of-experimental-covid-19-treatment-using/?utm_source=Nature+Briefing&utm_campaign=97ef9daa48-briefing-dy-20200407&utm_medium=email&utm_term=0_c9dfd39373-97ef9daa48-44480057) 

> All patients who are hospitalized and need oxygen will be able to sign up for the trial.

> “It’s a flip of a coin – two-thirds of patients will get the antibody-rich blood, with the other third being a control group,” Dr. Arnold said. The control group would receive standard care.

> The trial could be completed in as little as three months. Dr. Arnold said they are realistically looking at six to 10 months before they have definitive results.

> “The challenge is to get to an answer as quickly as possible so that we can start using this product if it’s truly effective, or realize it’s not and stop using it,” Dr. Arnold said.

<!--chapter:end:07-clinicaltrials.Rmd-->


# Resources

## Data Sources

- [JHU COVID19 time series data](https://github.com/CSSEGISandData/COVID-19)

- [New York Times COVID time series (USA state and county level data only)](https://github.com/nytimes/covid-19-data)

- [Bangladesh Unofficial deaths data](https://bit.ly/3em6TqO)

## Computational Resources

- R package [EpiModel](http://www.epimodel.org/)
- [Tim Churches Blog](timchurches.github.io/)
- [Learning Machines Blog](https://blog.ephorie.de/)
- [Infectious disease modeling beyond SIR model](https://towardsdatascience.com/infectious-disease-modelling-beyond-the-basic-sir-model-216369c584c4)

## Dashboards

- [Johns Hopkins Corona Virus Daboard](https://coronavirus.jhu.edu/map.html)

- http://91-divoc.com/pages/covid-visualization/

- [Beautiful Shiny Dashboard by Shubhram Pandey ](https://shubhrampandey.shinyapps.io/coronaVirusViz/_w_6ea524e3/_w_f14b34f2/)

## Dashboards from Bangladesh

- [By Rana Tuhin](https://rana2hin.shinyapps.io/covid19_bd/)

<!--chapter:end:20-resources.Rmd-->

`r if (knitr:::is_html_output()) '
# References {-}
'`

<!--chapter:end:30-references.Rmd-->

